src/sentry/seer/signed_seer_api.py

```diff
@@ -40,10 +43,15 @@ def make_signed_seer_api_request(
 
 def sign_with_seer_secret(url: str, body: bytes):
     auth_headers: dict[str, str] = {}
-    if random() < options.get("seer.api.use-shared-secret") and settings.SEER_API_SHARED_SECRET:
-        signature_input = b"%s:%s" % (url.encode("utf8"), body)
-        signature = hmac.new(
-            settings.SEER_API_SHARED_SECRET.encode("utf-8"), signature_input, hashlib.sha256
-        ).hexdigest()
-        auth_headers["Authorization"] = f"Rpcsignature rpc0:{signature}"
+    if random() < options.get("seer.api.use-shared-secret"):
+        if settings.SEER_API_SHARED_SECRET:
+            signature_input = b"%s:%s" % (url.encode("utf8"), body)
+            signature = hmac.new(
+                settings.SEER_API_SHARED_SECRET.encode("utf-8"), signature_input, hashlib.sha256
+            ).hexdigest()
+            auth_headers["Authorization"] = f"Rpcsignature rpc0:{signature}"
+        else:
+            logger.warning(
```

Comment:
should we just throw an exception?

Reply:
i guess warning is fine

Reply:
Yes I would prefer keeping it as a warning as the call may or may not fail based on the Seer side configuration.